docs/handcraft.rst

```diff
@@ -0,0 +1,2049 @@
+    }
+    
+
+fulfillments
+^^^^^^^^^^^^
+At the moment, we have to distinguish between fulfillment payloads **before**
+fulfillment and **after** fulfillment. For signature-based schemes, this means,
+before signature(s), and after signature(s).
+
+Moreover payloads for ``CREATE`` operations are a bit different.
+
+.. note:: We hope to be able to simplify the payload strucutre and validation,
+    such that this is no longer require.
+
+.. todo:: Point to issues addressing the topic.
+
+Example of a fulfillment payload **before fulfilling it**, for a `CREATE`
+operation:
+
+.. ipython::
+
+    fulfillment = {
+        'fid': 0,
+        'fulfillment': None,
+        'input': None,
+        'owners_before': ['8L6ngTZ5ixuFEr1GiunrFNWtGkft4swWWArXjWJu2Uwc'],
+    }
+
+.. note:: Because it is a `CREATE` operation, the ``'input'`` field is set to
+    ``None``.
+
+.. todo:: Example of a fulfillment payload **after fulfilling it**:
+
+
+.. _bicycle-asset-creation-revisited:
+
+********************************
+Bicycle Asset Creation Revisited
+********************************
+
+The Prepared Transaction
+========================
+Recall that to prepare a transaction we had to do something similar to:
+
+.. ipython::
+
+    In [0]: from bigchaindb_driver.crypto import generate_keypair
+
+    In [0]: from bigchaindb_driver.offchain import prepare_transaction
+
+    In [0]: alice = generate_keypair()
+
+    In [0]: bicycle = {
+       ...:     'data': {
+       ...:         'bicycle': {
+       ...:             'serial_number': 'abcd1234',
+       ...:             'manufacturer': 'bkfab',
+       ...:         },
+       ...:     },
+       ...: }
+
+    In [0]: metadata = {'planet': 'earth'}
+
+    In [0]: prepared_creation_tx = prepare_transaction(
+       ...:     operation='CREATE',
+       ...:     owners_before=alice.verifying_key,
+       ...:     asset=bicycle,
+       ...:     metadata=metadata,
+       ...: )
+
+and the payload of the prepared transaction looked similar to:
+
+.. ipython::
+    
+    In [0]: prepared_creation_tx
+
+Note ``alice``'s publci key:
+
+.. ipython::
+
+    In [0]: alice.verifying_key
+
+We are now going to craft this payload by hand.
+
+Extract asset id and metadata id:
+
+.. ipython::
+
+    In [0]: asset_id = prepared_creation_tx['transaction']['asset']['id']
+
+    In [0]: metadata_id = prepared_creation_tx['transaction']['metadata']['id']
+
+
+The transaction body
+--------------------
+
+asset
+^^^^^
+
+.. ipython::
+
+    In [0]: asset = {
+       ...:     'data': {
+       ...:         'bicycle': {
+       ...:             'manufacturer': 'bkfab',
+       ...:             'serial_number': 'abcd1234',
+       ...:         },
+       ...:     },
+       ...:     'divisible': False,
+       ...:     'refillable': False,
+       ...:     'updatable': False,
+       ...:     'id': asset_id,
+       ...: }
+
+.. note:: In a signature-less system, the values for the keys:
```

Comment:
Ah ok. The goal was to outline some key differences for developers who are familiar with HTTP server - client architectures in which the payloads can be very small since the server takes care of filling in default values.

But it can be removed.